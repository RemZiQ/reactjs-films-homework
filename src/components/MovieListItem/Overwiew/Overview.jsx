import React from 'react';
import propTypes from 'prop-types';

import MainInfo from '../MainInfo';
import './overview.scss';


export const Overview = ({
  id,
  overview,
  title,
  mark,
  genres,
  handleClose,
  handleForFetchTrailer,
  handleForOpenModal,
}) => {
  const handleForTrailer = (e) => {
    handleClose();
    handleForFetchTrailer(e);
    handleForOpenModal();
  };
  return (
    <div className="movieList__item__overview" id={`overview${id}`}>
      <button onClick={handleClose} type="submit" className="movieList__item__closeOverview" id={`closeOverview${id}`}>
        <i className="fas fa-times" />
      </button>
      <MainInfo title={title} mark={mark} genres={genres} />
      <div className="movieList__item__overviewDetails">{overview}</div>
      <button onClick={handleForTrailer} type="button" className="action__watch overview__watch" id={id}>Watch now</button>
    </div>
  );
};


Overview.propTypes = {
  id: propTypes.number,
  title: propTypes.string,
  mark: propTypes.number,
  genres: propTypes.arrayOf(propTypes.number),
  overview: propTypes.string,
  handleClose: propTypes.func,
  handleForFetchTrailer: propTypes,
  handleForOpenModal: propTypes,
};

Overview.defaultProps = {
  title: 'Sorry here is no title',
  mark: 1,
  overview: 'Sorry here is no info',
  id: null,
  genres: [],
  handleClose: () => {},
  handleForFetchTrailer: () => {},
  handleForOpenModal: () => {},
};

export default Overview;
