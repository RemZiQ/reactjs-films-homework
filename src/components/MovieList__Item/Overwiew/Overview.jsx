import React from 'react';
import MainInfo from '../MovieListItemMainInfo/MovieListItemMainInfo';

import './overview.scss';


const Overview = ({
  id, overview, title, mark, genres, handleClose,
}) => (
  <div className="movieList__item__overview" id={`overview${id}`}>
    <button onClick={handleClose} type="submit" className="movieList__item__closeOverview" id={`closeOverview${id}`}>
      <i className="fas fa-times" />
    </button>
    <MainInfo title={title} mark={mark} genres={genres} />
    <div className="movieList__item__overviewDetails">{overview}</div>
    <button type="button" className="action__watch overview__watch">Watch now</button>
  </div>
);

export default Overview;
