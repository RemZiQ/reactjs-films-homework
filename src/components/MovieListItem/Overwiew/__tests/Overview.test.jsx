import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { Overview } from '../Overview';


it('Overview renders well', () => {
  const renderer = new ShallowRenderer();
  const result = renderer.render(<Overview />);
  expect(result).toMatchSnapshot();
});
