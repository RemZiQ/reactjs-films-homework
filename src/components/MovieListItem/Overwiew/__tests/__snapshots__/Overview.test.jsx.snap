// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview renders well 1`] = `
<div
  className="movieList__item__overview"
  id="overviewnull"
>
  <button
    className="movieList__item__closeOverview"
    id="closeOverviewnull"
    onClick={[Function]}
    type="submit"
  >
    <i
      className="fas fa-times"
    />
  </button>
  <Connect(MainInfo)
    genres={Array []}
    mark={1}
    title="Sorry here is no title"
  />
  <div
    className="movieList__item__overviewDetails"
  >
    Sorry here is no info
  </div>
  <button
    className="action__watch overview__watch"
    id={null}
    onClick={[Function]}
    type="button"
  >
    Watch now
  </button>
</div>
`;
